geometry/create_capacitance_file_theta.py

```diff
@@ -0,0 +1,349 @@
+    hCapTrace[i].SetLineWidth(2)
+    hCapTrace[i].SetTitle("Stripline capacitance; #theta; Capacitance [pF]")
+    hCapTrace[i].SetName("hCapacitance_traces"+str(i))
+    #shields
+    hCapShield.append(TH1F())
+    hCapShield[i].SetBins(numTheta, minTheta, maxTheta)
+    hCapShield[i].SetLineColor(line_color_number)
+    hCapShield[i].SetLineStyle(line_style_number)
+    hCapShield[i].SetLineWidth(2)
+    hCapShield[i].SetTitle("Signal pads - ground shields capacitance; #theta; Capacitance [pF]")
+    hCapShield[i].SetName("hCapacitance_shields"+str(i))
+    #area
+    hCapDetector.append(TH1F())
+    hCapDetector[i].SetBins(numTheta, minTheta, maxTheta)
+    hCapDetector[i].SetLineColor(line_color_number)
+    hCapDetector[i].SetLineStyle(line_style_number)
+    hCapDetector[i].SetLineWidth(2)
+    hCapDetector[i].SetTitle("Signal pad - absorber capacitance; #theta; Capacitance [pF]")
+    hCapDetector[i].SetName("hCapacitance_detector"+str(i))
+    if line_color_number > 8:
+        line_color_number += 10
+    else:
+        line_color_number += 1
+    line_style_number += 1
+
+cTrace = TCanvas("cTrace","",600,400)
+cShield = TCanvas("cShield","",600,400)
+cDetector = TCanvas("cDetector","",600,400)
+
+legend = TLegend(0.1,0.693,0.8,0.9)
+#legend = TLegend(0.135,0.693,0.8,0.892)
+legend.SetHeader("Longitudinal layers")
+legend.SetNColumns(4)
+capa_shield_max = 0
+capa_det_max = 0
+for i in range (0, len(readoutLayerParallelLengths)):
+    print("--------------")
+    for index in range(0, numTheta):
+        theta = minTheta + index * deltaTheta
+        thetaNext = minTheta + (index+1) * deltaTheta
+        eta = -log(tan(theta/2.0))
+        deltaEta = abs( -log(tan((minTheta + (index+1) * deltaTheta)/2.0)) - eta)
+        if (debug):
+            print("theta = ", theta)
+            print("eta = ", eta)
+            print("delta eta = ", deltaEta)
+                         
+        # take into account the inclination in theta
+        traceLength = trace_length[i] / sin(theta)
+        #print("Layer %d trace length %f"%(i+1, traceLength))
+        #Trace capacitance (stripline) - not used since already accounted for elsewhere
+        logStripline = log(3.1 * hs / (0.8 * w + t))
+        # analytical formula
+        capacitanceTrace = nmultTrace * 1 / inch2mm * 1.41 * epsilonR / logStripline * traceLength
+        hCapTrace[i].SetBinContent(index+1, capacitanceTrace)
+    
+        #Shield capacitance (microstrip)
+        cellLength = readoutLayerParallelLengths[i] / sin(theta)
+        logMicrostrip = log(5.98 * hm / (0.8 * ws + t))
+        # analytical formula (nmultShield = 2)
+        #capacitanceShield = nmultShield * nMergedModules[i] * cellLength * tracesPerLayer[i] * 1 / inch2mm * 0.67 * (epsilonR + 1.41) / logMicrostrip
+        # from maxwell (nmultShield = 1)
+        capacitanceShield = nmultShield * nMergedModules[i] * nMergedThetaCells[i] * cellLength * tracesPerLayer[i] * capa_per_mm[i]
```

Comment:
Hi Giovanni, nice! I think there is indeed some double counting here: the number of merged module in theta here account for the fact that we simulate the strip size everywhere, then merge them to get the nominal cells. But the shield/pad capa is reasonably independent of the cell size and this part of the merging is already taken into account by the `tracesPerLayer`.

Reply:
Thanks for spotting this, so the bottom line is that the `nMergedThetaCells[i]`should actually not be applied for the shield/pad capacitance

Reply:
fixed in latest commit